### PR TITLE
Shortwave redistribution option added to namelist.

### PR DESCRIPTION
**Pull request by dabail10**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
We need the "shortwave redistribution" option for ktherm=2 for CESM. I believe it was needed for E3SM as well. So, I have added it here, but with an option to turn
it off via the namelist.
- [X] Developer(s): 
dabail10 (D. Bailey)
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
https://github.com/CICE-Consortium/Test-Results/wiki/icepack_by_mach_forks#izumi
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:
This is bfb with sw_redist=.true. and ktherm=1 and sw_redist = .false. and ktherm=2. Since the default for sw_redist will be .false. we should warn people who use the BL99 thermo (ktherm=1) that this will now be off and they will need to turn it on if necessary for their runs.

### CODE REVIEW

**Comment by apcraig** [comment]
Looks like some hardwired parameters were moved to namelist and the results are bit-for-bit.    Why are the alt03, alt04, and thermo1 test cases bit-for-bit with the code changes?

**Comment by dabail10** [comment]
For those tests, I added sw_redist = .true.

**Comment by dabail10** [comment]
This also helps get rid of some CESMCOUPLED ifdefs.

File: doc/source/science_guide/sg_thermo.rst [hunk 1]
```diff
@@ -50,6 +50,12 @@ the :ref:`thermo-growth` section. We begin by describing the melt ponds and sur
 forcing parameterizations, which are closely related to the ice and snow
 surface temperatures.
 
+Sometimes instabilities can arise when the temperature is close to the melt point in
+the snow and sea ice and there is abundant internal shorwave absorbed. One can chose
```
Comment (eclare108213) [inline doc/source/science_guide/sg_thermo.rst:54]: typo:  chose should be choose

**Comment by apcraig** [comment]
> For those tests, I added sw_redist = .true.

I'm still a little confused.  I understand that in alt03, alt04, and thermo1, sw_redist is now true.  What I don't understand is why everything is bit-for-bit with the prior results when sw_redist is true for some cases and false for other cases.  Was another flag (which one?) playing the same role as sw_redist prior to this change?  

As a followup, subroutine temperature_changes_salinity has some new code that is on when sw_redist is true.  Is this just diagnostic?

**Comment by dabail10** [comment]
The old code was only present in ice_therm_bl99.F90 (ktherm=1). It was not present in ice_therm_mushy.F90 (ktherm=2). So, it was only effectively "true" for the ktherm=1 case.

File: doc/source/science_guide/sg_thermo.rst (hunk 1 above)
Comment (dabail10) [reply doc/source/science_guide/sg_thermo.rst]: Fixed.

**Comment by apcraig** [comment]
I think all comments have been addressed and this is ready to merge.

**Pull request merged**